### ISSUE TITLE
Adding LCWSTR support to ::Lang::Win32

### ISSUE THREAD

**Issue by ikegami** [PerlFFI/FFI-Platypus#291]
Hi,

I'd like to add support for `LPCWSTR` (const Windows Unicode string) to FFI::Platypus.

There are two options for doing so
- A standalone module.
- Integration into FFI::Platypus::Lang::Win32.

As a standalone module, it would look like [this](https://gist.github.com/ikegami/9f9a0d83d920f36ab0daeed142e3392a). (I would add docs, and I would convert it to the coding style used by FFI::Platypus.) 

As-is, `lstrlenW` is very inefficient. I could replace it with 

```perl
use FFI::CheckLib qw( find_lib_or_die );
use FFI::Platypus qw( );
my $kernel32_ffi = FFI::Platypus->new( api => 1 );
$kernel32_ffi->lib(find_lib_or_die( lib => 'kernel32' ));
$kernel32_ffi->attach('lstrlenW', [ 'opaque' ], 'int');
```

But I would prefer to integrate it into FFI::Platypus::Lang::Win32 for two reasons.

1. Cleaner for the user to have the function in `->lang('Win32')`.
1. I could write a helper in XS to replace the very inefficient core of `native_to_perl`.

I spoke to Graham Ollis and he seemed ameable to both the idea as a whole and the XS approach.

**Comment by ikegami** [comment]
> It would be helpful if you can provide some examples of C code or some libraries where one might use this, just for discussion.

Every system call that accepts a string takes the string as a `LCWSTR`. For example, the following is used to create a file handle in Windows:

```c
HANDLE CreateFileW(
  LPCWSTR               lpFileName,
  DWORD                 dwDesiredAccess,
  DWORD                 dwShareMode,
  LPSECURITY_ATTRIBUTES lpSecurityAttributes,
  DWORD                 dwCreationDisposition,
  DWORD                 dwFlagsAndAttributes,
  HANDLE                hTemplateFile
);
```

[This](https://gist.github.com/ikegami/fb5ec425cca7b86eac72c04d4b49af4e) is a working example to get the parsed command line.

**Comment by ikegami** [comment]
Re "Help Wanted" tag, I have no problems finishing up the work and submitting a pull request. Slowly :) I created the issue to give a chance for comment.

**Comment by plicease** [comment]
This is addressed in 1.35_01.  I expect a production release in a week or so if there are no problems revealed from cpantesters.
